pyccel/parser/base.py

```diff
@@ -500,17 +503,118 @@ def insert_import(self, expr):
                         container[source] = []
                     container[source] += name
 
-    def print_namespace(self):
+    def dump(self, filename=None):
+        """
+        Dump the current ast using Pickle.
 
-        # TODO improve spacing
+          Parameters
+          ----------
+          filename: str
+            output file name. if not given `name.pyccel` will be used and placed
+            in the Pyccel directory ($HOME/.pyccel)
+        """
+        import pickle
+        import hashlib
+
+        if not filename:
+            if not self.filename:
+                raise ValueError('Expecting a filename to load the ast')
+
+            path , name  = os.path.split(self.filename)
+
+            if not name.split('.')[-1] == 'pyh':
+                return
+            else:
+                name = name[:-4]
 
-        print ('------- namespace -------')
-        for (k, v) in self.namespace.items():
-            print ('{var} \t :: \t {dtype}'.format(var=k, dtype=type(v)))
-        print ('-------------------------')
+            name     = '{}.pyccel'.format(name)
+            filename = path+ '/' + name
+        # check extension
 
-    def _visit(self, expr, **settings):
-        raise NotImplementedError('Must be implemented by the extension')
+        if not filename.split(""".""")[-1] == 'pyccel':
+            raise ValueError('Expecting a .pyccel extension')
+
+#        print('>>> home = ', os.environ['HOME'])
+        # ...
+
+        # we are only exporting the AST.
+        try:
+            code = self.code.encode('utf-8')
+            hs   = hashlib.md5(code)
+            f    = open(filename, 'wb')
+            pickle.dump((hs.hexdigest(), __version__, self), f, pickle.HIGHEST_PROTOCOL)
+            f.close()
+        except:
+            return
+
+    # TODO shall we need to load the Parser too?
```

Comment:
It is not very useful to have this comment here, which looks more like a reminder to self than a proper TODO item. I suggest we have a brief discussion on the topic instead.

Reply:
I forgot to delete it, it's not relevant anymore